### PR TITLE
Add retries for S3 file manipulation

### LINKED ISSUE #45: Retry failed S3 uploads
Occasionally S3 uploads fail for no apparent reason (https://github.com/aodn/issues/issues/150). If this happens, wait a bit and retry.

### PR DESCRIPTION
**Pull request by pmbohm**
fixes #45

### CODE REVIEW

File: aodncore/util/external/awsretry.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import logging
+import re
+import time
+from functools import wraps
+
+import boto
+import botocore
+
+__author__ = 'Allen Sanabria'
+__version__ = '1.0.2'
```
Comment (mhidas) [inline aodncore/util/external/awsretry.py]: Should there be a link to the source of this code here?

Comment (lwgordonimos) [reply aodncore/util/external/awsretry.py]: It's already here: https://github.com/aodn/python-aodncore/pull/61/files#diff-c9d6c5c00c596b6bbd4fbc00452260aaR5

I didn't want to modify the source unnecessarily, so we could (in theory, but unlikely) upgrade it in place. It is MIT licensed, so GPLv3 compatible, but if we really had to replace it, not a big deal either.

Comment (mhidas) [reply aodncore/util/external/awsretry.py]: :+1: Ok, that makes sense.

Comment (lwgordonimos) [reply aodncore/util/external/awsretry.py]: Want to make sure we're good open source citizens ;) Also, have fixed the merge conflict, which was caused by https://github.com/aodn/python-aodncore/pull/57

**Comment by mhidas** [comment]
@lwgordonimos Any idea why the Travis tests are failing? It says:
```
AttributeError: 'module' object has no attribute 'test_basestep'
```

**Comment by lwgordonimos** [comment]
Updated to fix tests... by changing library again because the other one was for some reason depending on both boto and boto3...

**Pull request merged**